.github/workflows/bench.yml

```
@@ -0,0 +1,358 @@
+    # to start a job whenever its repo has work. GitHub's `concurrency:` cannot
+    # coordinate across repos, so a filesystem flock mutex
+    # (tools/ci/with-nashua-lock.sh) keeps only one heavy job running at a time;
+    # it wraps the bench command in the "Run CodSpeed benchmarks" step below.
+    # See docs/ci/self-hosted-runner.md for what the box must provide: CodSpeed's
+    # own patched valgrind (NOT the distro valgrind package), libc6-dbg, flock and
+    # a fixed CPU model — node and yarn are provisioned per-job below. That doc
+    # also covers how the shared mutex works and the cutover steps.
+    # IMPORTANT: moving the bench between workflow files (or runners) is a
+    # baseline re-seed event: one main run must complete here before PR
+    # comparisons are meaningful again.
+    runs-on: [self-hosted, codspeed-bench, nashua]
```

actionlint (1.7.12)

[error] label "codspeed-bench" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2025-vs2026", "windows-2022", "windows-11-arm", "ubuntu-slim", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "macos-latest", "macos-latest-xlarge", "macos-latest-large", "macos-26-intel", "macos-26-xlarge", "macos-26-large", "macos-26", "macos-15-intel", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xlarge", "macos-14-large", "macos-14", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file

(runner-label)